### ISSUE TITLE
`jetpack.find('.')` fails on bad junction symlink

### ISSUE THREAD

**Issue by Dimava** [szwacz/fs-jetpack#104]
I was tydying up my storage when I met this error
```js
  jp = require('fs-jetpack');
  jp.write('./target/file.txt', 'wasd');
  console.log('good tree:', jp.inspectTree('.'));
  fs.symlinkSync('./target', './link', 'junction'); // create junction
  child_process.execSync('rmdir target /S /Q'); // remove './target'. Makes './link' invalid
  console.log('bad tree:', jp.inspectTree('.'));
  jp.find('.') // crashes
  // Uncaught TypeError: Cannot read properties of undefined (reading 'type')
  //     at ...\node_modules\fs-jetpack\lib\utils\tree_walker.js:71:20
  //     at Array.forEach (<anonymous>)
  //     at walkSync (...\node_modules\fs-jetpack\lib\utils\tree_walker.js:39:51)
  //     at Object.initialWalkSync [as sync] (...\node_modules\fs-jetpack\lib\utils\tree_walker.js:86:7)
  //     at findSync (...\node_modules\fs-jetpack\lib\find.js:80:14)
  //     at Object.findSyncInit [as sync] (...\node_modules\fs-jetpack\lib\find.js:116:10)
  //     at Object.find (...\node_modules\fs-jetpack\lib\jetpack.js:138:19)
```

**Comment by szwacz** [comment]
Thanks for reporting, will look into it next week.

**Comment by szwacz** [comment]
Fixed on v4.3.1
